modules/indexer/code/elasticsearch/elasticsearch.go

```diff
@@ -179,12 +179,38 @@ func (b *Indexer) Index(ctx context.Context, repo *repo_model.Repository, sha st
 		reqs = append(reqs, b.addDelete(filename, repo))
 	}
 
+	// Helper function to support Go v1.2 and below
+	min := func(a, b int) int {
+		if a <= b {
+			return a
+		}
+		return b
+	}
+
 	if len(reqs) > 0 {
-		_, err := b.inner.Client.Bulk().
-			Index(b.inner.VersionedIndexName()).
-			Add(reqs...).
-			Do(ctx)
-		return err
+		esBatchSize := 12 // Hardcoded batch size for ElasticSearch index update
+		var batchHead int
+
+		maxPerBatchReqCount := len(reqs)/esBatchSize + (len(reqs) % esBatchSize)
```

Comment:
I think it's totally wrong. It's very simple after you defined esBatchSize. Send requests every 12 of reqs.

Reply:
Well! I divided the whole reqs in 12 chunks rather than the other way. Will do a PR soon